### PR TITLE
feat: Add Deep connector Smartlead (Metadata)

### LINKED ISSUE #911: Add Smartlead deep connector

### PR DESCRIPTION
**Pull request by Cobalt0s**
Closes #911

# Description

Static schema file is served as Object Metadata.
The file is created using documentation. For campaigns the docs had less fields than what endpoint returns, so it was manually added.

# Testing
Verified campaings, client endpoints.
image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/909?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#909** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/909?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#906** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/906?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#901** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/901?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#902](https://github.com/amp-labs/connectors/pull/902) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/902?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#900** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/900?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#899** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/899?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#898** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/898?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: smartlead/metadata/scrapping.go
Comment (RajatPawar) [inline smartlead/metadata/scrapping.go:1]: This should be named `scraping.go`

File: smartlead/metadata/schemas.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
+{
+  "data": {
+    "campaigns": {
+      "displayName": "Campaigns",
+      "fields": {
+        "id": "id",
+        "user_id": "User id",
+        "created_at": "Create at",
```
Comment (RajatPawar) [inline smartlead/metadata/schemas.json:8]: This should be Created at, and `User id` can be `User ID`, same with client ID. Please update this wherever applicable for all objects (uppercase for acronyms)

File: test/smartlead/metadata/main.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	metadata, err := conn.ListObjectMetadata(ctx, []string{
+		objectName,
+	})
+	if err != nil {
+		utils.Fail("error listing metadata for Smartlead", "error", err)
+	}
+
+	slog.Info("Compare object metadata against endpoint response:")
```
Comment (RajatPawar) [inline test/smartlead/metadata/main.go:50]: Couple of minor typos - `Comparing`, `Success: Object metadata schema and endpoint response have the same fields`

**Review by RajatPawar — APPROVED** [review]
Mostly looks good, please resolve the comments before merging

File: smartlead/metadata/schemas.json (hunk 1 above)
Comment (Cobalt0s) [reply smartlead/metadata/schemas.json:8]: I was talking with Lauren the other day about ID conventions in golang code. And to be safe I made it lower case here.
Maybe for display names it is fine.

**Pull request merged**